route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -236,85 +284,61 @@ namespace route_following_plugin
+            maneuver_msg.lane_following_maneuver.end_time = start_time + ros::Duration(min_plan_duration_);
         } else {
-            maneuver_msg.lane_following_maneuver.end_time = current_time + ros::Duration((end_dist - current_dist) / (0.5 * cur_plus_target));
+            maneuver_msg.lane_following_maneuver.end_time = start_time + ros::Duration((end_dist - start_dist) / (0.5 * cur_plus_target));
         }
         maneuver_msg.lane_following_maneuver.lane_id = std::to_string(lane_id);
-        current_time = maneuver_msg.lane_following_maneuver.end_time;
-        ROS_INFO_STREAM("Creating lane follow start dist:"<<current_dist<<" end dist:"<<end_dist);
+        ROS_INFO_STREAM("Creating lane follow start dist:"<<start_dist<<" end dist:"<<end_dist);
         ROS_INFO_STREAM("Duration:"<<maneuver_msg.lane_following_maneuver.end_time.toSec() - maneuver_msg.lane_following_maneuver.start_time.toSec());
         return maneuver_msg;
     }
 
-    cav_msgs::Maneuver RouteFollowingPlugin::composeLaneChangeManeuverMessage(double current_dist, double end_dist, double current_speed, double target_speed, int starting_lane_id,int ending_lane_id, ros::Time& current_time){
+    cav_msgs::Maneuver RouteFollowingPlugin::composeLaneChangeManeuverMessage(double start_dist, double end_dist, double start_speed, double target_speed, int starting_lane_id,int ending_lane_id, ros::Time start_time){
         cav_msgs::Maneuver maneuver_msg;
         maneuver_msg.type = cav_msgs::Maneuver::LANE_CHANGE;
         maneuver_msg.lane_change_maneuver.parameters.neogition_type = cav_msgs::ManeuverParameters::NO_NEGOTIATION;
         maneuver_msg.lane_change_maneuver.parameters.presence_vector = cav_msgs::ManeuverParameters::HAS_TACTICAL_PLUGIN;
         maneuver_msg.lane_change_maneuver.parameters.planning_tactical_plugin = lane_change_plugin_;
         maneuver_msg.lane_change_maneuver.parameters.planning_strategic_plugin = "RouteFollowingPlugin";
-        maneuver_msg.lane_change_maneuver.start_dist = current_dist;
-        maneuver_msg.lane_change_maneuver.start_speed = current_speed;
-        maneuver_msg.lane_change_maneuver.start_time = current_time;
+        maneuver_msg.lane_change_maneuver.start_dist = start_dist;
+        maneuver_msg.lane_change_maneuver.start_speed = start_speed;
+        maneuver_msg.lane_change_maneuver.start_time = start_time;
         maneuver_msg.lane_change_maneuver.end_dist = end_dist;
         maneuver_msg.lane_change_maneuver.end_speed = target_speed;
         // because it is a rough plan, assume vehicle can always reach to the target speed in a lanelet
-        double cur_plus_target = current_speed + target_speed;
+        double cur_plus_target = start_speed + target_speed;
         if (cur_plus_target < 0.00001) {
-            maneuver_msg.lane_change_maneuver.end_time = current_time + ros::Duration(mvr_duration_);
+            maneuver_msg.lane_change_maneuver.end_time = start_time + ros::Duration(min_plan_duration_);
```

Comment:
why is the length of the entire maneuver plan used in this case. It seems like this is basically an error case and it would be better to log a warning and provide some small value like 1s instead of 16s for the bad maneuver. Same applies to the other method like this.

Reply:
I'm guessing the intention here was that if the current and target speed are zero, we should not move, so we just send an entire plan with that 0 speed

Reply:
That seems like a plausible reason, but such a case really should be handled by stop and wait and would therefore be an error for a lane following or lane change maneuver.

Reply:
That makes sense to me. But also at this point the vehicle is already stopped, so there is no way for route following to switch to a different maneuver (since the current implementation compares against downtrack to advance maneuvers). Should this be an exception still? I'm thinking about cases where we try to engage in areas with complete zero speed limit

Reply:
Any lanelet with a speed limit of zero really should be considered a bug in the map but I think my point stands. These methods are specific to lane change and lane follow both of which would inherently require non-zero speed since those maneuver requires motion by definition. So providing either method with 0 speed is an error.